stdlib/pagerduty/pagerduty.flux

```diff
@@ -0,0 +1,93 @@
+// `dedupkey` - string - a per alert ID. It acts as deduplication key, that allows you to ack or change the severity of previous messages.
+// `class` - string - The class/type of the event, for example ping failure or cpu load.
+// `group` - string - Logical grouping of components of a service, for example app-stack
+// `severity` - string - The perceived severity of the status the event is describing with respect to the affected system. This can be critical, error, warning or info.
+// `component` - string - Component of the source machine that is responsible for the event, for example mysql or eth0.
+// `source` - string - The unique location of the affected system, preferably a hostname or FQDN.
+// `summary` - string - A brief text summary of the event, used to generate the summaries/titles of any associated alerts. The maximum permitted length of this property is 1024 characters.
+// `timestamp` - string - The time at which the emitting tool detected or generated the event, in RFC 3339 nano format/
+sendEvent = (pagerdutyURL=defaultURL,
+    token="",
+    routingKey,
+    client,
+    clientURL,
+    dedupKey,
+    class,
+    group,
+    severity,
+    component,
+    source,
+    summary,
+    timestamp) => {
+
+    payload = {
+            summary: summary,
+            timestamp: timestamp,
+            source: source,
+            severity: severity,
+            component: component,
+            group: group,
+            class: class,
+    }
+    data = {
+        payload: payload,
+        routing_key: routingKey,
+        dedup_key: dedupKey,
+        event_action: actionFromSeverity(severity: severity),
+        client: client,
+        client_url: clientURL,
+    }
+
+    headers = {
+        "Authorization": "Token token=" + token,
+        "Accept": "application/vnd.pagerduty+json;version=2",
+        "Content-Type": "application/json",
+    }
+    enc = json.encode(v: data)
+    return http.post(headers: headers, url: pagerdutyURL, data: enc)
+}
+
+// `endpoint` creates the endpoint for the Slack external service.
+// `url` - string - URL of the slack endpoint. Defaults to: "https://slack.com/api/chat.postMessage", if one uses the webhook api this must be acquired as part of the slack API setup, and this URL will be secret.
+// `token` - string - token for the slack endpoint.  This can be ignored if one uses the webhook url acquired as part of the slack API setup, but must be supplied if the chat.postMessage API is used.
+// The returned factory function accepts a `mapFn` parameter.
+// The `mapFn` parameter must be a function that returns an object with `pagerdutyURL`, `token`, `client`, `client_url`,`class`,`group`, `severity`, `component`, `source`, `summary`, and `timestamp` as defined in the sendEvent function.
+// Note that while sendEvent accepts a dedup key, endpoint gets the dedupkey from the groupkey of the input table instead of it being handled by the `mapFn`.
+endpoint = (url=defaultURL, token="") =>
```

Comment:
This comment appears to be a copy paste from Slack, can you update it to be pagerduty specific?